### PR TITLE
Add index.js, change index.html, fix Error 405, images from Blobstore stored to Datastore.

### PR DESCRIPTION
**Pull request by Daljin**

### CODE REVIEW

File: currentIssue.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+There's an error 404 when you are redirected to commentPage.html
+Error 405 HTTP method GET is not supported by this URL - Whenever you submit.
```
Comment (sbahaud) [inline currentIssue.txt]: I thought this issue was fixed

File: currentIssue.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+There's an error 404 when you are redirected to commentPage.html
+Error 405 HTTP method GET is not supported by this URL - Whenever you submit. 
+index.html is a temporary solution to an issue that's causing an older version of index.html to be previewed instead of index.jsp.
+index.html is only redirecting user to index.jsp
```
Comment (sbahaud) [inline currentIssue.txt]: Not sure if you should submit this file. You can maintain in locally as a list of issues that needs to be resolved or add a TODO: in the relevant file saying that there is a known issue which will be resolved in a follow up PR.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -44,17 +52,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    persistMessage(request.getParameter("emailInput"), request.getParameter("messageInput"));
+    String imageUrl = getUploadedFileUrl(request, "image");
+    persistMessage(imageUrl, request.getParameter("message"));
 
-    response.sendRedirect("commentPage.html");
+    response.sendRedirect("/commentPage.html");
```
Comment (sbahaud) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Shouldn't the "/" be removed?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -63,18 +72,48 @@ private void persistMessage(String email, String message) {
...
+      String comment = (String) entity.getProperty("message");
       if (comment != null) {
         allComments.add(comment);
       }
     }
 
     return allComments;
   }
+
+  // From the example.
```
Comment (sbahaud) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Update the comment to something relevant to the method or remove it.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -63,18 +72,48 @@ private void persistMessage(String email, String message) {
...
+  private String getUploadedFileUrl(HttpServletRequest request, String image) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get("image");
+
+    // User submitted form without selecting a file, so we can't get a URL. (devserver)
```
Comment (sbahaud) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Is the "(devserver)" in the comment a mostake?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -63,18 +72,48 @@ private void persistMessage(String email, String message) {
...
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // We could check the validity of the file here, e.g. to make sure it's an image file
+    // https://stackoverflow.com/q/10779564/873165
```
Comment (sbahaud) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Does this comment add value? If not you can remove it.

File: currentIssue.txt (hunk 1 above)
Comment (Daljin) [reply currentIssue.txt]: This was, I'll be removing the text file as it no longer needed

File: currentIssue.txt (hunk 2 above)
Comment (Daljin) [reply currentIssue.txt]: Oh, right! I can also add it to the projects tab by creating a kanban with tasks to do and issues to solve.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (Daljin) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: The "/" should be there as without will cause issues in the test site.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (Daljin) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done, the comment was made there to show it's from an example. I don't know if we are supposed to use the examples code or make our own, but it's counterintuitive to make our own code, so I think its fine.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (Daljin) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I think it adds values to me, not so much to others. It's there so I can learn to restrict images. I'll remove it and add it onto a text file in my local computer

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (Daljin) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I don't think so, I may not understand the question.

**Pull request merged**